### PR TITLE
express-notes-api - Lamberto Martinez

### PR DESCRIPTION
**Pull request by Lambpato**

### CODE REVIEW

**Comment by Lambpato** [comment]
my bad, will provide those after this exercise

**Comment by Lambpato** [comment]
Screenshot 2023-03-22 at 11 21 48 AM
Screenshot 2023-03-22 at 11 15 32 AM
Screenshot 2023-03-22 at 11 13 53 AM
Screenshot 2023-03-22 at 11 16 08 AM
Screenshot 2023-03-22 at 11 19 44 AM

File: express-notes-api/server.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import express from 'express';
+
+import { readFile, writeFile } from 'node:fs/promises';
+
+const app = express();
+const json = JSON.parse(await readFile('data.json'));
+let message = {};
```
Comment (RobertGardner) [inline express-notes-api/server.js:7]: This shouldn't be a global variable. Just declare it inside each function right before you set it.

File: express-notes-api/server.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+app.use(express.json());
+
+app.get('/api/notes', (req, res) => {
+  try {
+    res.json(Object.values(json.notes));
+    res.status(200);
```
Comment (RobertGardner) [inline express-notes-api/server.js:14]: It's more common to chain these: `res.status().json()`. Like you did below when sending a message.

Also, 200 is the default so you don't actually even need this call.

File: express-notes-api/server.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    }
+  } catch (err) {
+    console.error(err.message);
+    res.status(500).send(err.message);
+  }
+});
+
+app.post('/api/notes', async (req, res) => {
+  try {
+    if (Object.keys(req.body).includes('content')) {
```
Comment (RobertGardner) [inline express-notes-api/server.js:43]: Or, more succinct: `if ('content' in req.body)`

**Review by RobertGardner — APPROVED** [review]
This is good code.

**Pull request merged**